### PR TITLE
TGUI Safes

### PR DESCRIPTION
**Pull request by blackrep**
## About the PR 
This PR refactors Secure Safe code to a more modern standard. As a part of the PR, the following changes have been made:
1. Replace magic 1s and 0s with `TRUE` and `FALSE`
2. Support variable safe code length (previously hardcoded to 4)
3. Fix bug where holding secure briefcases would not show boop / beep hints
4. Add sounds when you press the keypad buttons
5. Convert safe UI to TGUI

## Why's this needed? 
The new UI is far more responsive, and clean.
| Old | New |
| ---- | ----- |
|  |  |

## Changelog

```changelog
(u)Blackrep
(+)Safes have a fancy new UI
```

### CODE REVIEW

File: tgui/packages/tgui/interfaces/SecureSafe.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  let content = padMsg ? padMsg : stylizeCode(attempt, codeLen);
+  if (disabled) {
+    content = "NO ACCESS";
+  }
+  if (emagged) {
+    content = glitch(content, 2);
+  }
+  return (
+    // Crude hack to get the styling I want
+    <ProgressBar
```
Comment (Luxizzle) [inline tgui/packages/tgui/interfaces/SecureSafe.js:44]: We shouldn't be abusing the progress bar component like this. It is fine to add a bit of custom CSS for your interface for whatever purpose this is doing.

File: code/obj/item/storage/secure_safe.dm [hunk 2]
```diff
@@ -1,387 +1,257 @@
...
-			overlays = null
-			boutput(usr, "<span class='alert'>The lock mechanism clunks locked.</span>")
-			src.visible_message("<span class='alert'>[src]'s lock mechanism clunks locked.</span>")
-			playsound(src.loc, "sound/items/Deconstruct.ogg", 65, 1)
-/*
-	else if (href_list["setcode"])
-		if (src.locked && src.code)
-			return
+				// We're dealing with a configured safe, try to open / close it using the set code
+				src.submit_guess()
```
Comment (ZeWaka) [inline code/obj/item/storage/secure_safe.dm:152]: pass `usr` into this proc as a `mob/user`

**Review by Luxizzle — CHANGES_REQUESTED** [review]

**Comment by blackrep** [comment]
Wow I really messed that up.

**Comment by blackrep** [comment]
Redoing this PR with #9776. I really messed up try to rebase this, and do not want to try to fix this.

**Pull request closed without merging**